index.js

```diff
@@ -49,27 +46,31 @@ module.exports = (opts) => {
 
       let body = ''
 
+      // Per RFC 7231 §3.1.1.5 if this header is not present we MAY assume application/octet-stream
+      const contentType = req.headers['content-type'] || 'application/octet-stream'
+
       // according to https://tools.ietf.org/html/rfc2616#section-4.3
-      // proxy should ignore message body when it's a GET or HEAD request
+      // fastify ignore message body when it's a GET or HEAD request
       // when proxy this request, we should reset the content-length to make it a valid http request
+      // discussion: https://github.com/fastify/fastify/issues/953
       if (req.method === 'GET' || req.method === 'HEAD') {
         headers['content-length'] = 0
       } else {
         if (req.body) {
           if (req.body instanceof Stream) {
             body = req.body
-          } else {
-            // Per RFC 7231 §3.1.1.5 if this header is not present we MAY assume application/octet-stream
-            const contentType = req.headers['content-type'] || 'application/octet-stream'
```

Comment:
This seemed important, considering the RFC ref.

Reply:
Its just set on a bit higher level. See line 55.